aiida/engine/processes/process.py

```diff
@@ -761,6 +763,55 @@ def _setup_inputs(self) -> None:
             elif isinstance(self.node, orm.WorkflowNode):
                 self.node.base.links.add_incoming(node, LinkType.INPUT_WORK, name)
 
+    def _setup_inputs_non_db(self) -> None:
+        """Store JSON-serializable values of ``non_db`` ports in the node's attributes."""
+        non_db_inputs = self._get_inputs_non_db_serializable(self.spec().inputs, self.raw_inputs)
+        pruned = prune_mapping(non_db_inputs)
+        self.node.set_non_db_inputs(pruned)
+
+    def _get_inputs_non_db_serializable(
+        self,
+        port: Union[None, InputPort, PortNamespace],
+        port_value: Any,
+    ):
+        """Return the inputs that correspond to ports with ``non_db=True`` and that are JSON serializable.
+
+        The function is called recursively for any port namespaces.
+
+        :return: A copy of ``port_value`` where all inputs that do no correspond to a non-db port or are not JSON
+            serializable, have been filtered out.
+        """
+        from aiida.orm.implementation.utils import clean_value
```

Comment:
is there a reason to have this import inside the function?

Reply:
Not really, moved it to module level